src/Config/ObjectModel/RuntimeConfig.cs

```diff
@@ -6,12 +6,60 @@
 
 namespace Azure.DataApiBuilder.Config.ObjectModel;
 
-public record RuntimeConfig(
-    [property: JsonPropertyName("$schema")] string Schema,
-    DataSource DataSource,
-    RuntimeOptions Runtime,
-    RuntimeEntities Entities)
+public record RuntimeConfig
 {
+    [JsonPropertyName("$schema")]
+    public string Schema;
+
+    public DataSource DataSource;
+
+    public RuntimeOptions Runtime;
+
+    public RuntimeEntities Entities;
+
+    [JsonIgnore]
+    public string DefaultDataSourceName;
+
+    [JsonIgnore]
+    public Dictionary<string, DataSource> DatasourceNameToDataSource { get; set; }
+
+    [JsonIgnore]
+    public Dictionary<string, string> EntityNameToDataSourceName { get; set; }
+
+    [JsonConstructor]
+    public RuntimeConfig(string Schema, DataSource DataSource, RuntimeOptions Runtime, RuntimeEntities Entities)
```

Comment:
this would be a breaking change for anyone using parameter names to reference this constructor: 
![image](https://github.com/Azure/data-api-builder/assets/124841904/90d34a47-fa9f-47b6-88f0-78616ca79381). Hence want to avoid it for this one.

Reply:
discussed offline. resovling.